### PR TITLE
[NewIR] Printer for Program/Operation/Type and Register ops for ResNet50

### PR DESCRIPTION
**Pull request by kangguangli**
### PR types

New features
### PR changes

Others
### Description

This PR does the following things:
1. add printer for Program/Operation/Type
2. register empty ops for ResNet50
#### Others
Pcard-67164

### CODE REVIEW

File: paddle/fluid/dialect/pd_op.h [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    static const char *name() { return OPNAME(op_name); } \
+    static const char **attributes_name;                  \
+    static constexpr uint32_t attributes_num = 0;         \
+  };                                                      \
+  const char **className::attributes_name = nullptr;
+
+REIGSTER_EMPTY_OP(conv2d, Conv2DOp);
+REIGSTER_EMPTY_OP(feed, FeedOp);
+REIGSTER_EMPTY_OP(batch_norm, BatchNormOp);
+REIGSTER_EMPTY_OP(batch_norm_, BatchNormOp_);
```
Comment (zhangbo9674) [inline paddle/fluid/dialect/pd_op.h:36]: BatchNormOp_ 是不是写成 BatchNorm_Op 更合理一些？

File: paddle/ir/printer.cc [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+#include "paddle/ir/builtin_attribute.h"
+#include "paddle/ir/builtin_type.h"
+#include "paddle/ir/dialect.h"
+#include "paddle/ir/operation.h"
+#include "paddle/ir/program.h"
+#include "paddle/ir/value.h"
+
+namespace ir {
+
+namespace {
```
Comment (zhangbo9674) [inline paddle/ir/printer.cc:29]: 这个 namespace 还需要添加么？

**Review by zhangbo9674 — APPROVED** [review]

File: paddle/ir/printer.cc (hunk 2 above)
Comment (kangguangli) [reply paddle/ir/printer.cc:29]: 这是用来让`newline`只在`printer.cc`里可见的，避免该变量暴露在`ir`命名空间中。
This is used to make `newline` only visible in `printer.cc` and avoid it exposed to `ir` namespace.

File: paddle/fluid/dialect/pd_op.h (hunk 1 above)
Comment (kangguangli) [reply paddle/fluid/dialect/pd_op.h:36]: 可以的，之后再改。
Sure, will change later.

**Review by winter-wang — APPROVED** [review]
LGTM

**Pull request merged**